frame/glutton/src/lib.rs

```diff
@@ -0,0 +1,258 @@
+			ensure_root(origin)?;
+
+			// Fill up the `TrashData` storage item.
+			(0..trash_count).for_each(|i| TrashData::<T>::insert(i, &[i as u8; 1024]));
+
+			Self::deposit_event(Event::PalletInitialized);
+			Ok(())
+		}
+
+		/// Set the `Compute` storage value that determines how much of the
+		/// block's weight `ref_time` to use during `on_idle`.
+		///
+		/// Only callable by Root.
+		#[pallet::call_index(1)]
+		#[pallet::weight(T::DbWeight::get().writes(1))]
+		pub fn set_compute(origin: OriginFor<T>, compute: Perbill) -> DispatchResult {
+			ensure_root(origin)?;
+			Compute::<T>::set(compute);
+
+			Self::deposit_event(Event::ComputationLimitSet { compute });
+			Ok(())
+		}
+
+		/// Set the `Storage` storage value that determines the PoV size usage
+		/// for each block.
+		///
+		/// Only callable by Root.
+		#[pallet::call_index(2)]
+		#[pallet::weight(T::DbWeight::get().writes(1))]
+		pub fn set_storage(origin: OriginFor<T>, storage: Perbill) -> DispatchResult {
+			ensure_root(origin)?;
+			Storage::<T>::set(storage);
+
+			Self::deposit_event(Event::StorageLimitSet { storage });
+			Ok(())
+		}
+	}
+
+	impl<T: Config> Pallet<T> {
+		/// Wastes some `proof_size`. Receives a counter as an argument.
+		pub(crate) fn waste_at_most_proof_size(meter: &mut WeightMeter) {
+			let Ok(n) = Self::calculate_proof_size_iters(&meter) else {
+				return;
+			};
+
+			meter.defensive_saturating_accrue(T::WeightInfo::waste_proof_size_some(n));
+
+			(0..n).for_each(|i| {
+				TrashData::<T>::get(i);
+			});
+		}
+
+		/// Calculate how many times `waste_proof_size_some` should be called to full up `meter`.
+		fn calculate_proof_size_iters(meter: &WeightMeter) -> Result<u32, ()> {
+			let base = T::WeightInfo::waste_proof_size_some(0);
+			let slope = T::WeightInfo::waste_proof_size_some(1).saturating_sub(base);
+
+			let remaining = meter.remaining().saturating_sub(base);
+			let iter_by_proof_size =
+				remaining.proof_size().checked_div(slope.proof_size()).ok_or(())?;
+			let iter_by_ref_time = remaining.ref_time().checked_div(slope.ref_time()).ok_or(())?;
+
+			match iter_by_proof_size {
+				0 => Err(()),
+				i if i <= iter_by_ref_time => Ok(i as u32),
+				_ => Err(()),
+			}
+		}
+
+		/// Waste at most the remaining ref time weight of `meter`.
+		///
+		/// Tries to come as close to the limit as possible.
+		pub(crate) fn waste_at_most_ref_time(meter: &mut WeightMeter) {
+			let Ok(n) = Self::calculate_ref_time_iters(&meter) else {
+				return;
+			};
+			meter.defensive_saturating_accrue(T::WeightInfo::waste_ref_time_iter(n));
+
+			let clobber = Self::waste_ref_time_iter(vec![0u8; 64], n);
+
+			// By casting it into a vec we can hopefully prevent the compiler from optimizing it
+			// out. Note that `Blake2b512` produces 64 bytes, this is therefore impossible - but the
+			// compiler does not know that (hopefully).
+			debug_assert!(clobber.len() == 64);
+			if clobber == vec![0u8; 65] {
```

Comment:
```suggestion
			if clobber.len() == 65 {
```

Should probably be enough.